api/middlewares/upload_middleware.go

```diff
@@ -146,7 +158,28 @@ func (u UploadMiddleware) Handle() gin.HandlerFunc {
 
 				uploadFileName, fileUID := u.randomFileName(conf, ext)
 				fileReader := bytes.NewReader(fileByte)
+
 				errGroup.Go(func() error {
+					if conf.WebpEnabled {
+						buff := new(bytes.Buffer)
+						options := &webp.Options{
+							Lossless: false,
+							Quality:  75,
+						}
+						var decodedImage image.Image
+						decodedImage, _, err = image.Decode(bytes.NewReader(fileByte))
+						if err != nil {
+							return err
+						}
+						if err := webp.Encode(buff, decodedImage, options); err != nil {
+							return err
+						}
+						uploadWebpFileName := u.bucketPath(conf, fmt.Sprintf("%s_webp%s", fileUID, ext))
+
+						if _, err := u.bucket.UploadFile(ctx, buff, uploadWebpFileName, strings.ReplaceAll(fileHeader.Filename, ext, "")+".webp"); err != nil {
+							return err
+						}
+					}
```

Comment:
@Denes-cilwal This implementation needs to be done in two goroutines. Separated into two error groups instead of doing it in the same error group. 

I want you to learn why we have used `errGroup` in this implementation. 
https://www.fullstory.com/blog/why-errgroup-withcontext-in-golang-server-handlers 
This link might help. 

Also, before converting into webp always need to check if the uploaded file is a supported image or not.

Reply:
> @Denes-cilwal This implementation needs to be done in two goroutines. Separated into two error groups instead of doing it in the same error group.
> 
> I want you to learn why we have used `errGroup` in this implementation.
> https://www.fullstory.com/blog/why-errgroup-withcontext-in-golang-server-handlers
> This link might help.
> 
> Also, before converting into webp always need to check if the uploaded file is a supported image or not.

@dipeshdulal  i am still unclear about this  last comments. Let discuss this tommorrow.